### ISSUE TITLE
The value of a component class differs depending on where it's referenced

### ISSUE THREAD

**Issue by levitanong** [omcljs/om#792]
In clj mode:

```clj
(defui Component
  static SomeProtocol
  (some-protocol [this]
    ;; returns some expected representation of Component, complete with meta
    (println Component))
  Object
  (render [this] 
    ;; returns some function without meta
    (println Component)))
```

**Comment by levitanong** [comment]
Just realized this is likely expected behavior, as Component in render method will refer to the instance rather than the class.

**Comment by anmonteiro** [comment]
Right, but printing the component instance could be more informative. Reopening until #796 is merged.

**Comment by levitanong** [comment]
I see. On that note, is there a way to get the class from an instance? Basically the reverse of `any->class`.

**Comment by levitanong** [comment]
Scratch that, I feel silly now! it's `react-type`!

**Comment by anmonteiro** [comment]
fixed 1b66d2b067fd31523b31e1c3a3a20449547e0fa2
